lib/msf/ui/console/command_dispatcher/modules.rb

```diff
@@ -1434,14 +1434,30 @@ def show_favorites # :nodoc:
               return
             end
 
-            # create module set using the saved modules
-            fav_modules = {}
-            saved_favs = File.readlines(favs_file)
-            saved_favs.each do |mod|
-              module_name = mod.strip
-              fav_modules[module_name] = framework.modules[module_name]
-            end
-            show_module_metadata('Favorites', fav_modules)
+            # get the full module names from the favorites file and use then to search the MetaData Cache for matching modules
+            saved_favs = File.readlines(favs_file).map(&:strip)
+            @module_search_results = Msf::Modules::Metadata::Cache.instance.find('fullname' => [saved_favs, []])
```

Comment:
Added a check that would print a warning if a module that is `favs_file` no longer exists in Framework.

```suggestion
            fav_modules.each do |fullname, mod_obj|
              if mod_obj.nil?
                print_warning("#{favs_file} contains a module that can not be found - #{fullname}.")
              end
            end

            @module_search_results = Msf::Modules::Metadata::Cache.instance.find('fullname' => [saved_favs, []])
```